src/core/qgscadutils.cpp

```diff
@@ -162,7 +162,10 @@ QgsCadUtils::AlignMapPointOutput QgsCadUtils::alignMapPoint( const QgsPointXY &o
 
   // *****************************
   // ---- Common Angle constraint
-  if ( numberOfHardLock < 2 && !ctx.angleConstraint.locked && ctx.cadPoints().count() >= 2 && ctx.commonAngleConstraint.locked && ctx.commonAngleConstraint.value != 0 )
+  if ( numberOfHardLock < 2 && !ctx.angleConstraint.locked && ctx.cadPoints().count() >= 2 && ctx.commonAngleConstraint.locked && ctx.commonAngleConstraint.value != 0
+       // Skip common angle constraint if the snapping to vertex/edge has priority
+       && ( !( snapMatch.hasVertex() || snapMatch.hasEdge() ) || ! ctx.snapToVertexOrEdgeHasPriorityOverCommonAngle )
```

Comment:
Why only vertex and segment? I believe all snapping modes should be treated the same for consistency, including _snapping on intersection_

Reply:
done